src/Error/ExceptionRenderer.php

```diff
@@ -191,10 +191,7 @@ public function render()
         $template = $this->_template($exception, $method, $code);
         $unwrapped = $this->_unwrap($exception);
 
-        $isDebug = Configure::read('debug');
-        if (($isDebug || $exception instanceof HttpException) &&
-            method_exists($this, $method)
-        ) {
+        if (method_exists($this, $method)) {
```

Comment:
This change will require us to document this in the migration guide and docs. In the past custom exception pages could be 'dev' only but now they are used in production mode too. I suspect there may be custom application error pages that are not unlike our own 'missing action' error pages, that emit far too much information for a production mode.

Reply:
True but with debug off only either `erro400` or `error500` template is used, irrespective of method called.

https://github.com/cakephp/cakephp/blob/7a7eed81bc3e32ccf3382b126a15d46369243473/src/Error/ExceptionRenderer.php#L314

Also additional exception attributes are set as view vars only when debug if off 

https://github.com/cakephp/cakephp/blob/7a7eed81bc3e32ccf3382b126a15d46369243473/src/Error/ExceptionRenderer.php#L230

So I don't think there's a risk of leaking any sensitive info. Yet to be extra safe I am fine moving this patch to 4.0 instead.

Reply:
If the templates are always overridden then we should be reasonably safe.